code.gs

```diff
@@ -2026,10 +2028,13 @@ function getEventsOnDate (eventDate, calendarId) {
     log.add('The calendar with ID "' + calendarId + '" is not accessible: check your calendarId value!', Priority.FATAL_ERROR);
   }
 
+  eventDate = dateWithTimezone(year, month, day, 0, 0, 0, eventCalendar.timeZone);
```

Comment:
I am not sure if I understood exactly what you are saying and  ince it was exactly an off-by-one error with an hour in a date variable which caused this problem, so this definitely needs to be addressed.

I jotted down this test function and it seems to work as expected, can you confirm this is the case?

```javascript  
function testDate () {
  // Winter: +0200 expected
  var date = new dateWithTimezone(2018, 10, 1, 0, 0, 0, 'Europe/Athens');
  Logger.log(Utilities.formatDate(date, 'Europe/Athens', 'yyyy-MM-dd\'T\'HH:mm:ssXXX'));
  // Logs '2018-11-01T00:00:00+02:00'

  // Summer: +0300 expected
  date = new dateWithTimezone(2018, 3, 1, 0, 0, 0, 'Europe/Athens');
  Logger.log(Utilities.formatDate(date, 'Europe/Athens', 'yyyy-MM-dd\'T\'HH:mm:ssXXX'));
  // Logs '2018-04-01T00:00:00+03:00'
}
```

Reply:
Yep, that was what I meant. I copy-pasted the test and it works as intended for me too. Sorry I was too lazy to implement a test myself (yet motivated enough to ask, so that you had to do it). :-D

Maybe it would be good to keep that test-function (or something similar) in the `tests.gs` file, to ensure nothing changes out from under us...